timemachine/cpp/src/kernels/k_translations.cuh

```diff
@@ -23,13 +22,20 @@ void __global__ k_generate_translations_within_or_outside_a_sphere(
     const RealType center_y = center[1];
     const RealType center_z = center[2];
 
-    curandState_t local_state;
-    local_state = rand_states[threadIdx.x];
+    curandState_t local_state = rand_states[threadIdx.x];
 
-    if (targeting_inner[0] == 1) {
-        while (idx < num_translations) {
-            // TBD: Whether or not this randomness needs to be double
-            // Done this way initially due to issues with ptxas
+    const RealType box_x = box[0 * 3 + 0];
+    const RealType box_y = box[1 * 3 + 1];
+    const RealType box_z = box[2 * 3 + 2];
+
+    const RealType inv_box_x = 1 / box_x;
+    const RealType inv_box_y = 1 / box_y;
+    const RealType inv_box_z = 1 / box_z;
+
+    while (idx < num_translations) {
+        // TBD: Whether or not this randomness needs to be double
+        // Done this way initially due to issues with ptxas
+        if (idx % 2 == 0) {
```

Comment:
was going to suggest some speed-ups but looks like this is basically a no-op in the profiling results

Reply:
Would be curious to hear the suggestions regardless, if we could speed this up we can speed up the tests

Reply:
This diverges the threads, better to have all `tid < total_num_threads//2` do inner, and `tid >= total_num_threads//2` do outer?